src/ott/tools/plot.py

```diff
@@ -245,3 +260,159 @@ def animate(
+  ):
+
+    super().__init__(
+        fig=fig,
+        ax=ax,
+        cmap=cmap,
+        scale_alpha_by_coupling=scale_alpha_by_coupling,
+        alpha=alpha,
+        title=title
+    )
+
+    self._patches = []
+    self._points = []
+    self._n, self._k, self._top_k = None, None, None
+
+  def __call__(self, ot: mmsinkhorn.MMSinkhornOutput) -> List["plt.Artist"]:
+    """Plot 2-D couplings. does not support higher dimensional."""
+    n_s = [len(ot.x_s[i]) for i in range(len(ot.x_s))]
+    assert self._n < jnp.prod(
+        jnp.array(n_s)
+    ), "Intended number of tuples too large."
+
+    # Extract top_k largest entries in the tensor, and their indices.
+    val, idx = jax.lax.top_k(ot.tensor.ravel(), self._top_k)
+    indices = jnp.unravel_index(idx, n_s)
+
+    # Setttings for plot
+    markers = "svopxdh"
+
+    for j in range(self._top_k):
+      points = [ot.x_s[i][indices[i][j], :] for i in range(self._k)]
+      points = [points[i] for i in ccworder(jnp.array(points))]
+      alphas = np.linspace(0.6, 0.2, self._top_k - self._n)
+      alpha = 0.6 if j < self._n else alphas[j - self._n]
+      points = ptc.Polygon(
+          points,
+          fill=True,
+          linewidth=2,
+          color=self._cmap[j > self._n],
+          alpha=alpha,
+          zorder=-j,
+      )
+      self._patches.append(self.ax.add_patch(points))
+
+    for i in range(self._k):
+      for j, val in enumerate(ot.x_s[i]):
+        self._points.append(
+            self.ax.scatter(
+                val[0],
+                val[1],
+                s=200 * ot.a_s[i][j] * len(ot.a_s[i]),
+                marker=markers[i],
+                c="black",
+                linewidth=0.0,
+                edgecolor=None,
+                label=str(i)
+            )
+        )
+
+    if self._title is not None:
+      self.ax.set_title(self._title)
+
+    return self._points + self._patches
+
+  def update(
+      self,
+      ot: mmsinkhorn.MMSinkhornOutput,
+      title: Optional[str] = None
+  ) -> List["plt.Artist"]:
+    """Update a plot with a transport instance."""
+    n_s = [len(ot.x_s[i]) for i in range(len(ot.x_s))]
+    assert self._n < jnp.prod(jnp.array(n_s)), \
+      "Intended number of tuples too large."
+    # Extract top_k largest entries in the tensor, and their indices.
+    val, idx = jax.lax.top_k(ot.tensor.ravel(), self._top_k)
+    indices = jnp.unravel_index(idx, n_s)
+
+    alphas = np.linspace(0.6, 0.2, self._top_k - self._n)
+    for j in range(self._top_k):
+      points = [ot.x_s[i][indices[i][j], :] for i in range(self._k)]
+      # reorder to ensure polygons have maximal area
+      points = [points[i] for i in ccworder(jnp.array(points))]
+      alpha = 0.6 if j < self._n else alphas[j - self._n]
+      self._patches[j].set_xy(points)
+      self._patches[j].set_color(self._cmap[j > self._n])
+      self._patches[j].set_alpha(alpha)
+
+    for i in range(self._k):
+      for j, val in enumerate(ot.x_s[i]):
```

Comment:
As mentioned above, if possible, I would avoid iterating over the points.

Reply:
not sure how. will think of it